src/profile/directory.mo

```diff
@@ -0,0 +1,64 @@
+import Array "mo:stdlib/array.mo";
+import HashMap "mo:stdlib/hashMap.mo";
+import Hash "mo:stdlib/hash.mo";
+import Iter "mo:stdlib/iter.mo";
+import List "mo:stdlib/list.mo";
+
+import Types "./types.mo";
+
+module {
+	type Profile = Types.Profile;
+	type PrincipalId = Types.PrincipalId;
+
+	public class Directory () {
+		func passthrough (hash : PrincipalId) : PrincipalId { hash };
+		let hashMap = HashMap.HashMap<PrincipalId, Profile>(1, Hash.Hash.hashEq, passthrough);
+
+		public func get (userId : PrincipalId) : ?Profile {
+			hashMap.get(userId)
+		};
+
+		public func set (userId : PrincipalId, _profile : Profile) : () {
+			let profile : Profile = {
+				id = userId;
```

Comment:
Storing the hashed blob as `userId` in the record itself, so it's easily usable by the front-end (without implementing crc32 there as well).